File: packages/android-sdk/src/main/kotlin/com/phosra/sdk/models/CompiledPolicy.kt
package com.phosra.sdk.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * The full compiled policy returned by the Phosra API at GET /device/policy.
 * This is the single source of truth for all parental controls to be enforced on-device.
 *
 * Maps exactly to Go service.CompiledPolicy struct.
 */
@Serializable
data class CompiledPolicy(
    @SerialName("version")
    val version: Int,

    @SerialName("child_id")
    val childId: String,

    @SerialName("child_age")
    val childAge: Int,

    @SerialName("age_group")
    val ageGroup: String,

    @SerialName("policy_id")
    val policyId: String,

    @SerialName("status")
    val status: String,

    @SerialName("generated_at")
    val generatedAt: String,

    @SerialName("content_filter")
    val contentFilter: ContentFilter,

    @SerialName("screen_time")
    val screenTime: ScreenTime,

    @SerialName("purchases")
    val purchases: Purchases,

    @SerialName("privacy")
    val privacy: Privacy,

    @SerialName("social")
    val social: Social,

    @SerialName("notifications")
    val notifications: PhosraNotifications,

    @SerialName("web_filter")
    val webFilter: WebFilter
)

File: packages/ios-sdk/Sources/PhosraSDK/Models/CompiledPolicy.swift
import Foundation

/// A fully compiled policy for on-device enforcement.
///
/// This struct matches the Go `service.CompiledPolicy` exactly. It is returned by
/// `GET /device/policy` and contains all rule settings organized by category.
public struct CompiledPolicy: Codable, Sendable {
    /// Policy version number. Increments each time a parent modifies a rule.
    public let version: Int

    /// UUID of the child this policy applies to.
    public let childID: String

    /// Child's current age in years.
    public let childAge: Int

    /// Age group label (e.g., "toddler", "child", "tween", "teen").
    public let ageGroup: String

    /// UUID of the parent-created policy.
    public let policyID: String

    /// Policy status ("active", "paused", "draft").
    public let status: String

    /// ISO 8601 timestamp when this policy was compiled.
    public let generatedAt: String

    /// Content filtering rules (age ratings, blocked/allowed apps).
    public let contentFilter: ContentFilter

    /// Screen time limits and schedules.
    public let screenTime: ScreenTime

    /// Purchase restriction settings.
    public let purchases: Purchases

    /// Privacy controls (location, profile visibility, data sharing).
    public let privacy: Privacy

    /// Social interaction controls (chat, DMs, multiplayer).
    public let social: Social

    /// Notification curfew and usage timer settings.
    public let notifications: Notifications

    /// Web content filtering rules.
    public let webFilter: WebFilter

    enum CodingKeys: String, CodingKey {
        case version, status, purchases, privacy, social, notifications
        case childID = "child_id"
        case childAge = "child_age"
        case ageGroup = "age_group"
        case policyID = "policy_id"
        case generatedAt = "generated_at"
        case contentFilter = "content_filter"
        case screenTime = "screen_time"
        case webFilter = "web_filter"
    }
}
